src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -764,7 +764,9 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
                 StandardUsernameCredentials credentials = CredentialsMatchers.firstOrNull(urlCredentials, idMatcher);
                 if (credentials != null) {
                     c.addCredentials(url, credentials);
-                    CredentialsProvider.track(project.getLastBuild(), credentials);
+                    if (project != null && project.getLastBuild() != null) {
```

Comment:
If `project` can really be null, should the parameter not be marked `@CheckForNull`?